java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -522,6 +615,23 @@ protected <T> T handleValkeyResponse(
             return null;
         }
 
+        // Handle DirectByteBuffer conversion for large responses (>16KB)
+        if (value instanceof java.nio.ByteBuffer) {
+            java.nio.ByteBuffer buffer = (java.nio.ByteBuffer) value;
+            byte[] bytes = new byte[buffer.remaining()];
```

Comment:
Is it possible to write a fast-path that avoids this? deserializeMapFromBytes() just wraps the byte array into a ByteBuffer again.